code/modules/paperwork/photography.dm

```diff
@@ -305,6 +315,9 @@
 	var/icon/res = get_base_photo_icon()
 
 	for(var/atom/A in plane_layer_sort(atoms))
+		iterator++
+		if(iterator % tickspersleep == 0)
+			sleep(1)
```

Comment:
Idea for this : `CHECK_TICK`, which makes it sleep only if the CPU used by the server is currently at or above a certain threshold.

Reply:
Tried this out and seems to work basically the same but in theory should be more smooth, so I changed it.